illegal instruction on android arm64

### Rollup Version

v4.46.1

### Operating System (or Browser)

Android ARM64

### Node Version (if applicable)

v24.4.0

### Link To Reproduction

_Not Available_

### Expected Behaviour

```text
$ pnpm exec rollup

rollup version 4.44.2
=====================================
...
```


### Actual Behaviour

```text
$ pnpm exec rollup
[ERR_PNPM_RECURSIVE_EXEC_FIRST_FAIL] Command was killed with SIGILL (Invalid machine instruction): rollup
```

--- a/rust/bindings_napi/Cargo.toml
+++ b/rust/bindings_napi/Cargo.toml
@@ -38,7 +38,7 @@ mimalloc-safe = { version = "0.1.54", features = ["local_dynamic_tls"] }
 # Disable architecture specific optimizations on aarch64 platforms
 # Mimalloc assumes `armv8.1-a` is supported, which is not always the case
 # Ref: https://github.com/rollup/rollup/issues/6047
-[target.'cfg(target_arch = "aarch64")'.dependencies]
+[target.'cfg(all(target_arch = "aarch64", not(target_vendor = "apple")))'.dependencies]
 mimalloc-safe = { version = "0.1.54", features = ["no_opt_arch"] }
 
 [build-dependencies]

--- a/rust/bindings_napi/Cargo.toml
+++ b/rust/bindings_napi/Cargo.toml
@@ -18,11 +18,28 @@ xxhash = { path = "../xxhash" }
 [target.'cfg(not(any(target_os = "linux", target_os = "freebsd")))'.dependencies]
 mimalloc-safe = { version = "0.1.54" }
 
-[target.'cfg(any(all(target_os = "linux", not(target_arch = "aarch64"), not(target_arch = "loongarch64"), not(all(target_arch = "riscv64", target_env = "musl"))), all(target_os = "freebsd", not(target_arch = "aarch64"))))'.dependencies]
+# Readable version of the target specifier
+# cfg(
+# 	any(
+# 		all(
+# 			target_os = "linux",
+# 			not(target_arch = "loongarch64"),
+# 			not(all(target_arch = "riscv64", target_env = "musl"))
+# 		),
+# 		all(
+# 			target_os = "freebsd",
+# 			not(target_arch = "aarch64")
+# 		)
+# 	)
+# )
+[target.'cfg(any(all(target_os = "linux", not(target_arch = "loongarch64"), not(all(target_arch = "riscv64", target_env = "musl"))), all(target_os = "freebsd", not(target_arch = "aarch64"))))'.dependencies]
 mimalloc-safe = { version = "0.1.54", features = ["local_dynamic_tls"] }
 
-[target.'cfg(all(target_os = "linux", target_arch = "aarch64"))'.dependencies]
-mimalloc-safe = { version = "0.1.54", features = ["local_dynamic_tls", "no_opt_arch"] }
+# Disable architecture specific optimizations on aarch64 platforms
+# Mimalloc assumes `armv8.1-a` is supported, which is not always the case
+# Ref: https://github.com/rollup/rollup/issues/6047
+[target.'cfg(target_arch = "aarch64")'.dependencies]
+mimalloc-safe = { version = "0.1.54", features = ["no_opt_arch"] }
 
 [build-dependencies]
 napi-build = "2.2.3"

